staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -363,18 +364,63 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 	if err != nil {
 		return err
 	}
-	c.AddHealthChecks(healthz.NamedCheck("etcd", func(r *http.Request) error {
+	c.AddHealthChecks(healthz.NamedGroupedCheck("etcd", "etcd", func(r *http.Request) error {
 		return healthCheck()
 	}))
 
 	readyCheck, err := storagefactory.CreateReadyCheck(s.StorageConfig, c.DrainedNotify())
 	if err != nil {
 		return err
 	}
-	c.AddReadyzChecks(healthz.NamedCheck("etcd-readiness", func(r *http.Request) error {
+	c.AddReadyzChecks(healthz.NamedGroupedCheck("etcd-readiness", "etcd-readiness", func(r *http.Request) error {
 		return readyCheck()
 	}))
 
+	if len(s.EtcdServersOverrides) != 0 {
+		if err := s.addOverrideEtcdHealthEndpoint(c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *EtcdOptions) addOverrideEtcdHealthEndpoint(c *server.Config) error {
+	uniqueServerSets := make(map[string]struct{})
+	for _, override := range s.EtcdServersOverrides {
+		tokens := strings.Split(override, "#")
+		servers := strings.Split(tokens[1], ";")
+		// multi overrides may point to the same servers
+		// example: ["apps/deployments#s1.example.com;s2.example.com","apps/replicasets#s1.example.com;s2.example.com"]
+		// multi overrides servers may in different order
+		// example: ["apps/deployments#s2.example.com;s1.example.com","apps/replicasets#s1.example.com;s2.example.com"]
+		sortedServers := make([]string, len(servers))
+		copy(sortedServers, servers)
+		sort.Strings(sortedServers)
+		sortedKey := strings.Join(sortedServers, ";")
```

Comment:
it's strange to use the sorted set here to determine uniqueness, but then use the unsorted set below to initialize the client ... if the order matters to the client, it should matter for uniqueness as well

I don't know enough to know if sort order matters, but we should at least be consistent ... use the same thing to determine uniqueness and client construction

Reply:
I changed to 
1. sort servers in ParseEtcdServersOverrides
2. check if overrides are dup in the next loop.

Reply:
> * sort servers in ParseEtcdServersOverrides

Does changing the specified order change client behavior? If so, I don't think we should sort before passing to the client.

Reply:
https://github.com/kubernetes/kubernetes/compare/406faa2db0b5ab6f2b6aac72a15175c3c9d37f15..6364526efb5072d71632556ac401d8f4f7b2d201

How about this? 
- keep the order in ParseEtcdServersOverrides
- make a copy for sorting and keep the order for the client